tests/benchmarks/test_xgboost.py

```diff
@@ -0,0 +1,237 @@
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+import pytest
+import xgboost.dask
+from dask_ml.metrics import mean_squared_error
+
+from ..utils_test import downscale_dataframe, run_up_to_nthreads, wait
+
+
+@pytest.fixture
+def taxi_zone_lookup():
+    # File from https://www.nyc.gov/site/tlc/about/tlc-trip-record-data.page
+    df = pd.read_csv(
+        "s3://coiled-runtime-ci/xgboost/taxi+_zone_lookup.csv",
+        usecols=["LocationID", "Borough"],
+    )
+    BOROUGH_MAPPING = {
+        "Manhattan": "Superborough 1",
+        "Bronx": "Superborough 1",
+        "EWR": "Superborough 1",
+        "Brooklyn": "Superborough 2",
+        "Queens": "Superborough 2",
+        "Staten Island": "Superborough 3",
+        "Unknown": "Unknown",
+    }
+    df["Superborough"] = [BOROUGH_MAPPING[k] for k in df["Borough"]]
+    df = df.astype({"Borough": "string[pyarrow]", "Superborough": "string[pyarrow]"})
+    return df
+
+
+@run_up_to_nthreads("small_cluster", 200, reason="fixed size dataset")
+def test_preprocess(small_client, taxi_zone_lookup, read_parquet_with_pyarrow):
+    """A typical workflow that preprocesses crude data into a ML-friendly dataframe"""
+    ############
+    # Read input
+    ############
+    ddf = dd.read_parquet(
```

Comment:
Is there a reason why here you are not casting the strings to pyarrow strings?

Reply:
I just saw the fixture `read_parquet_with_pyarrow`. We should put a comment saying that we are doing this.

Reply:
Reading the strings as string[python] and then casting them manually to string[pyarrow] would cost you a huge memory spike which, for the 830MiB chunks, you can't afford without killing off the workers.
Adding a comment.